rust/blockstore/src/arrow/provider.rs

```diff
@@ -62,6 +80,29 @@ impl ArrowBlockfileProvider {
         }
     }
 
+    pub async fn prefetch(&self, id: &Uuid) -> Result<usize, ArrowBlockfileProviderPrefetchError> {
+        let block_ids = self
```

Comment:
Why not just fetch the root and then do this? The abstractions feel a bit off to me here.

Reply:
Fetching the root requires K to be known, which would break a general-purpose prefetch operator that has no knowledge of the concrete K/V types for each file path.

Reply:
ah that sucks, might be nice to comment that